### PR TITLE
Add custom GitHub Copilot agents for WP Rocket

### PR DESCRIPTION
**Pull request by Miraeld**
# Description

Fixes #xx
Doesn't impact users. 
Adds copilot agents for us to use.

## Type of change

- [x] Chore

### What was tested

Nothing to be tested

### How to test

Nothing to test

### Documentation

This pull request adds new agent definition files for WP Rocket, each outlining clear responsibilities, boundaries, and best practices for specialized automation roles in documentation, grooming, linting, performance, and testing. These agent specifications will help standardize contributions, improve code quality, and streamline the development workflow.

**Agent definitions and responsibilities:**

* **Documentation Agent**
  - Introduces `docs_agent` for PHPDoc documentation, specifying standards, required tags, file scope, and boundaries for public API documentation.

* **Grooming Agent**
  - Adds `grooming_agent` for technical issue refinement and planning, detailing the grooming process, output formats for user stories and bugs, and architectural patterns relevant to WP Rocket.

* **Lint Agent**
  - Defines `lint_agent` for enforcing WordPress Coding Standards, including critical style rules, allowed exceptions, and strict boundaries to prevent logic changes or editing third-party code.

* **Performance Agent**
  - Specifies `performance_agent` for building and maintaining optimization features, with guidance on CSS/JS, caching, database routines, background jobs, and performance patterns.

* **Test Agent**
  - Establishes `test_agent` for writing comprehensive PHPUnit tests, outlining conventions, required annotations, and boundaries for test contributions.

# Mandatory Checklist

## Code validation

- [x] I validated all the Acceptance Criteria. If possible, provide screenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I implemented built-in tests to cover the new/changed code.

## Code style

- [ ] I wrote a self-explanatory code about what it does.
- [ ] I protected entry points against unexpected inputs.
- [ ] I did not introduce unnecessary complexity.
- [ ] Output messages (errors, notices, logs) are explicit enough for users to understand the issue and are actionnable.

## Unticked items justification
There is no code style.

### CODE REVIEW

File: .github/agents/docs-agent.agent.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+- **Tech Stack:** PHP 7.3+, WordPress 5.8+, PHPDoc standards
+- **Plugin Type:** WordPress caching and performance optimization plugin
+- **Documentation Style:** WordPress documentation standards
+- **File Structure:**
+  - `inc/` – Plugin source code (you READ and WRITE here)
+  - `inc/Engine/` – Core engine components
+  - `inc/classes/` – Legacy classes
+  - `inc/Addon/` – Add-on features
+  - `inc/ThirdParty/` – Third-party compatibility
+
```
Comment (jeawhanlee) [inline .github/agents/docs-agent.agent.md:32]: Should we not add the views too?

File: .github/agents/docs-agent.agent.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+## Commands you can use
+Check documentation: `composer phpcs -- --sniffs=Squiz.Commenting,WordPress.Commenting`
+Check all code standards: `composer phpcs`
+Auto-fix style: `composer phpcs:fix`
+
```
Comment (jeawhanlee) [inline .github/agents/docs-agent.agent.md:38]: What about phpstan?

File: .github/agents/docs-agent.agent.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+ */
+function rocket_clean_domain( $domain ) {
+    // Missing @since, @param, @return
+}
+```
+
+## Boundaries
+- ✅ **Always do:** Add PHPDoc to public classes/methods/functions in `inc/`, include `@since` version tag, document parameters with types, document return values, keep descriptions under 100 chars per line, validate with `composer phpcs`, match documentation to actual implementation
+- ⚠️ **Ask first:** Changing existing documentation that affects public API understanding, documenting deprecated functions (ensure proper `@deprecated` tag), adding complex array parameter documentation with many nested `@type` definitions
+- 🚫 **Never do:** Modify code logic while documenting, add inaccurate documentation, document private implementation details that may change, skip `@param` or `@return` tags for public methods, use wrong text domain (must be `rocket`), commit without running `composer phpcs`
```
Comment (jeawhanlee) [inline .github/agents/docs-agent.agent.md]: What about phpstan here too?

File: .github/agents/grooming-agent.agent.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+- Understand the problem deeply: read issue descriptions, acceptance criteria, and related code
+- Reproduce bugs and identify root causes by analyzing codebase
+- Brainstorm multiple solution approaches and recommend the best one
+- Identify potential side-effects, risks, and dependencies
+- Break down work into clear development steps (sub-tasks)
+- Estimate effort using t-shirt sizing (XS/S/M/L/XL)
+- Write technical specifications with implementation guidance
+- Suggest refactoring opportunities when beneficial
+- Identify test scenarios and acceptance criteria gaps
+- Never implement the solution - only plan and document it
```
Comment (jeawhanlee) [inline .github/agents/grooming-agent.agent.md:21]: I think we should also add that grooming needs to be kept simple and precise too.

File: .github/agents/grooming-agent.agent.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+- Use `@group` annotations appropriately
+
+## Common pitfalls to avoid in solutions
+
+- ❌ Proposing `add_action`/`add_filter` directly → ✅ Must use Subscriber pattern
+- ❌ Synchronous API calls on page load → ✅ Use background jobs
+- ❌ Modifying existing database without migration → ✅ Version upgrades
+- ❌ Missing nonce verification in AJAX → ✅ Always check nonces
+- ❌ Not using type hints → ✅ Add type hints for PHP 7.3+
+- ❌ Using `apply_filters()` → ✅ Use `wpm_apply_filters_typed()`
```
Comment (jeawhanlee) [inline .github/agents/grooming-agent.agent.md:157]: What about suggesting ajax endpoints for new feature implementation, use REST instead

File: .github/agents/test-agent.agent.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  - `inc/classes/` – Legacy classes
+  - `inc/Addon/` – Add-on features (Cloudflare, etc.)
+  - `inc/ThirdParty/` – Third-party compatibility
+  - `tests/Unit/` – Unit tests (you WRITE here)
+  - `tests/Integration/` – Integration tests (you WRITE here)
+  - `tests/Fixtures/` – Test fixtures and mock data
+
+## Commands you can use
+Run unit tests: `composer test-unit` or `php -d memory_limit=512M vendor/bin/phpunit --configuration tests/Unit/phpunit.xml.dist --testdox`
+Run integration tests: `composer test-integration` (requires WordPress test environment)
```
Comment (jeawhanlee) [inline .github/agents/test-agent.agent.md:36]: Can we add that for integration test, we can first run bin/install-wp-tests.sh wordpress_test root root to install WP test environment.

File: .github/agents/test-agent.agent.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+Run specific test file: `php -d memory_limit=512M vendor/bin/phpunit --testdox <path-to-test-file>`
+Run tests with coverage: `composer test-unit-coverage`
+Check code style: `composer phpcs`
+
```
Comment (jeawhanlee) [inline .github/agents/test-agent.agent.md:41]: What about phpstan?

File: .github/agents/test-agent.agent.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+- `@group Cloudflare`, `@group WithWoo` - Third-party integrations
+- `@group Preload`, `@group RUCSS` - Feature-specific
+
+**Naming:**
+- Test class: `Test_{ClassName}` 
+- Test method: `testShouldDescribeExpectedBehavior`
+- Always use `@covers` with fully qualified class name
+
+## Boundaries
+- ✅ **Always do:** Write tests to `tests/Unit/` or `tests/Integration/`, follow naming conventions (`Test_ClassName`), use `@covers` and `@group` annotations, run tests with `composer test-unit` before submitting, test edge cases (empty, null, invalid input)
```
Comment (jeawhanlee) [inline .github/agents/test-agent.agent.md:64]: What about stating that test path should correspond to the one in `inc`

**Review by jeawhanlee — APPROVED** [review]
